### PR TITLE
change line in evaluate train seg

### LINKED ISSUE #493: multiclass loss computation error for validation and test
This [line](https://github.com/NRCan/geo-deep-learning/blob/develop/train_segmentation.py#L441) in the evaluation function of training pipeline gives error for multiclass loss calculation for losses such as cross entropy.
It should change to 
`loss = criterion(outputs, labels) if num_classes > 1 else criterion(outputs, labels.unsqueeze(1).float())`
The same way as it is computed in the training function.

### PR DESCRIPTION
**Pull request by CharlesAuthier**
fix #493 and add test for losses

### CODE REVIEW

File: train_segmentation.py [hunk 1]
```diff
@@ -438,7 +438,7 @@ def evaluation(eval_loader,
                                    ep_num=ep_idx + 1,
                                    scale=scale)
 
-            loss = criterion(outputs, labels.unsqueeze(1).float())
+            loss = criterion(outputs, labels) if num_classes > 1 else criterion(outputs, labels.unsqueeze(1).float())
```
Comment (ms888ekb) [inline train_segmentation.py:441]: have you tested that this line fails for the multiclass evaluation? 
In my tests loss = criterion(outputs, labels.unsqueeze(1).float()) works for both cases

Comment (CharlesAuthier) [reply train_segmentation.py:441]: yeah I created a test for it, it dont crash for all binary but some yeah and it pass for the news tests for binary and multiclass loss function that have a yaml in config

File: config/dataset/test_ci_segmentation_binary.yaml [hunk 2]
```diff
@@ -15,4 +15,4 @@ dataset:
   class_name: # will follow in the next version
   classes_dict: {'BUIL':1}
   class_weights:
-  ignore_index: 255
+  ignore_index: -1
```
Comment (ms888ekb) [inline config/dataset/test_ci_segmentation_binary.yaml]: why ignore_index = -1? it is  a convention to use 255 for 8-bit data.

Comment (CharlesAuthier) [reply config/dataset/test_ci_segmentation_binary.yaml]: yeah but it was like that before so I rechange it to the original value

Comment (ms888ekb) [reply config/dataset/test_ci_segmentation_binary.yaml]: I think that 255 is the correct way to mark ignor_index pixels.
Let's hear what other mates think

**Review by ms888ekb — APPROVED** [review]

**Review by valhassan — APPROVED** [review]
